fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/RdfStreamStreamingOutput.java

```diff
@@ -128,7 +130,21 @@ private static void write(final RdfStream rdfStream,
             final StreamRDF stream = getWriterStream(output, format);
             stream.start();
             nsPrefixes.forEach(stream::prefix);
-            rdfStream.forEach(stream::triple);
+
+            // ensure that writes to the stream object are serialized
+            // in case the rdfStream object is running in parallel mode.
+            final ReentrantLock lock = new ReentrantLock();
+            rdfStream.forEach(x -> {
+                try {
+                    lock.tryLock(30, TimeUnit.SECONDS);
```

Comment:
@ajs6f :  This seems like a good idea to me.  @awoods:  any reason not to go the subclassing route?